### PR TITLE
Update xgboost learner to latest CRAN version

### LINKED ISSUE #45: Enable more xgboost parameters
What is it about [this TODO](https://github.com/mlr-org/mlr3learners/blob/6dc9ef64d7811b8be26ffab6ce1febca8c99dfd4/R/LearnerClassifXgboost.R#L60)? If this worked with mlr it should also work with mlr3? 

Please be aware that the `tree_method` `default` is *wrong* as far as I understand [the documentation of xgboost](https://xgboost.readthedocs.io/en/latest/parameter.html): The default should be `"auto"`.

### PR DESCRIPTION
**Pull request by pat-s**
- add the already existing (new) parameters of the latest CRAN version of xgboost
- add parameter dependencies

In addition, some small fixes which including #30 	

fixes #45

### CODE REVIEW

**Comment by mb706** [comment]
tree_method default should be "auto" I believe: https://xgboost.readthedocs.io/en/latest/parameter.html

Also the inequality condition could be done with `CondAnyOf`.

**Comment by be-marc** [comment]
All missing parameters are added. Some parameters have a complex dependency on the `updater` parameter. This might be impossible to check with `paradox` at the moment. However, `xgboost` prints the possible combinations if something is wrong.

**Comment by be-marc** [comment]
Closes #45

**Comment by pat-s** [comment]
👍 

> This might be impossible to check with paradox at the moment.

Did you open an issue or similar? Couldn't find anything in the repo.

In addition, once accepted, could you please reflect those change also to mlr2?

File: NEWS.md [hunk 1]
```diff
@@ -9,6 +9,9 @@
 * Improved documentation.
 * Added references.
 
+* add new parameters of xgboost version 0.90.2
+* add parameter depenencies for xgboost
```
Comment (pat-s) [inline NEWS.md]:
```suggestion
* add parameter dependencies for xgboost
```

**Comment by be-marc** [comment]
If the `booster` parameter is set to `gbtree` or `dart`, `updater` must be a comma separated string  which can contain the following words `"grow_colmaker", "distcol", "grow_histmaker", "grow_local_histmaker", "grow_skmaker", sync", "refresh", "prune"`.  If the `booster` parameter is set to `gblinear`, `updater` must be a string containing `shotgun` or `coord_descent`. 

 We can check this with the following code

```
if(x$booster == "gblinear") {
  if(!x$updater %in%  c("shotgun", "coord_descent")) {
     # Error or replace by default value?
   }
  } else if (x$booster == "gbtree" || x$booster == "dart") {
    split_paramter = strsplit(x$updater, ",")
    split_paramter = lapply(split_paramter, function(x) gsub(" ", "", x))
    if(!all(split_paramter %in% c("grow_colmaker", "distcol", "grow_histmaker",
                                                   "grow_local_histmaker", "grow_skmaker",
                                                    "sync", "refresh", "prune"))) {
     # Error or replace by default value?
   }
  }
```
So `updater` is not a really tune-able parameter. This is more about preventing the user from inserting a wrong combination.  Am I right that this is not possible with `add_dep` since it depends on the values in both parameters? We can include it as a `trafo` function. How do I apply the `trafo` before invoking the training?

**Comment by pat-s** [comment]
AFAIR we agreed on taking the `updater` arg easy as it won't be used that often and people using it should be aware of what's required.
Is that correct?

@be-marc Maybe we can finish up this PR in the next days.

**Comment by pat-s** [comment]
@be-marc some tests are failing. Could you take a look?

**Comment by be-marc** [comment]
@pat-s The failing tests are resolved in #46

**Comment by pat-s** [comment]
Thanks again @be-marc!

**Pull request merged**